olp-cpp-sdk-dataservice-read/src/DownloadItemsJob.h

```diff
@@ -0,0 +1,146 @@
+};
+
+template <typename ItemType, typename PrefetchResult>
+class DownloadItemsJob {
+ public:
+  DownloadItemsJob(DownloadFunc download,
+                   Callback<PrefetchResult> user_callback,
+                   PrefetchStatusCallback status_callback)
+      : download_(std::move(download)),
+        user_callback_(std::move(user_callback)),
+        status_callback_(std::move(status_callback)) {}
+
+  void Initialize(size_t items_count, client::NetworkStatistics statistics) {
+    download_task_count_ = total_download_task_count_ = items_count;
+    accumulated_statistics_ = statistics;
+  }
+
+  ExtendedDataResponse Download(const std::string& data_handle,
+                                client::CancellationContext context) {
+    return download_(data_handle, context);
+  }
+  size_t GetAccumulatedBytes(const olp::client::NetworkStatistics& statistics) {
+    // This narrow cast is necessary to avoid narrowing compiler errors like
+    // -Wc++11-narrowing when building for 32bit targets.
+    const auto bytes_transferred =
+        statistics.GetBytesDownloaded() + statistics.GetBytesUploaded();
+    return static_cast<size_t>(bytes_transferred &
+                               std::numeric_limits<size_t>::max());
+  }
+
+  void CompleteItem(ItemType item, ExtendedDataResponse response) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    accumulated_statistics_ += GetNetworkStatistics(response);
+
+    if (response.IsSuccessful()) {
+      prefetch_result_.push_back(
+          std::make_shared<typename PrefetchResult::value_type::element_type>(
+              item,
+              typename PrefetchResult::value_type::element_type::ResultType()));
+      requests_succeeded_++;
+    } else {
+      prefetch_result_.push_back(
+          std::make_shared<typename PrefetchResult::value_type::element_type>(
+              item, response.GetError()));
+      requests_failed_++;
+    }
+
+    if (status_callback_) {
+      status_callback_(
+          PrefetchStatus{prefetch_result_.size(), total_download_task_count_,
+                         GetAccumulatedBytes(accumulated_statistics_)});
+    }
+
+    if (!--download_task_count_) {
+      OLP_SDK_LOG_DEBUG_F("PrefetchJob",
+                          "Download complete, succeeded=%zu, failed=%zu",
+                          requests_succeeded_, requests_failed_);
+
+      OnPrefetchCompleted(std::move(prefetch_result_));
+    }
+  }
+
+  void OnPrefetchCompleted(Response<PrefetchResult> result) {
+    auto prefetch_callback = std::move(user_callback_);
+    prefetch_callback(std::move(result));
+  }
+
+ private:
+  DownloadFunc download_;
+  Callback<PrefetchResult> user_callback_;
+  PrefetchStatusCallback status_callback_;
+
+  size_t download_task_count_{0};
+  size_t total_download_task_count_{0};
+
+  size_t requests_succeeded_{0};
+  size_t requests_failed_{0};
+
+  client::NetworkStatistics accumulated_statistics_;
+
+  PrefetchResult prefetch_result_;
+
+  std::mutex mutex_;
```

Comment:
Can you please remove the newlines between the members. There is no reason to separate the members in groups.